### ISSUE TITLE
clip_gradient with clip_grad_value

### ISSUE THREAD

**Issue by dhkim0225** [PyTorchLightning/pytorch-lightning#5456]
## 🚀 Feature

Same issue with #4927 
The current clip_gradient uses clip_grad_norm; can we add clip_grad_value?

https://github.com/PyTorchLightning/pytorch-lightning/blob/f2e99d617f05ec65fded81ccc6d0d59807c47573/pytorch_lightning/plugins/native_amp.py#L63-L65

**Comment by tchaton** [comment]
Hey @dhkim0225,

Sounds like a great feature. Would you mind making a PR to enable it ?

Best regards,
T.C

**Comment by dhkim0225** [comment]
@tchaton I'm really glad to do that !

Renaming the `clip_grad_value` variable to `clip_grad_norm` and,
adding a new `clip_grad_value` feature seems most intuitive, 
but this is BC-breaking. 

Do you have any good ideas?

Sincerely,
Anthony Kim.

**Comment by tchaton** [comment]
Hey @dhkim0225,

I just checked. You can pass those arguments directly to the Trainer.
trainer = Trainer(gradient_clip_val=10, track_grad_norm=2) and they will be used in NativeAMP clip_gradient

**Comment by priancho** [comment]
Hi @tchaton,

There are two popular gradient clipping methods: one that limits the maximum gradient value of each model parameter and the other one that scales the gradient value based on the p-norm of a (sub-)set of model parameters.

PyTorch Lightning implements the second option which can be used with Trainer's `gradient_clip_val` parameter as you mentioned.
This clipping algorithm is useful when the norm of gradients is large, but not when a small set of model parameters have abnormal gradient values since the norm will still be reasonably small considering the number of all model parameters.
I think that @dhkim0225 wants an alternative gradient clipping method that uses `torch.nn.utils.clip_grad_value_()` instead of `torch.nn.utils.clip_grad_norm_()` for such a scenario.

By the way, I don't think that this functionality is something that can break BC.
What about to add `gradient_clip_algorithm` parameter to Trainer, which is by default 'norm' but can be set to 'value'?
